### ISSUE TITLE
Container name clash when deploying a mixed protocol load balancer service

### ISSUE THREAD

**Issue by tomruttle** [k3s-io/k3s#4652]
**Environmental Info:**
K3s Version: 
```
k3s version v1.21.5+k3s2 (724ef700)
go version go1.16.8
```

Node(s) CPU architecture, OS, and Version: k3os v0.21.5-k3s2r1
```
Linux katana 5.4.0-88-generic #99 SMP Tue Oct 5 16:53:38 UTC 2021 x86_64 GNU/Linux
```

Cluster Configuration:
1 server, 0 agents

**Describe the bug:**
The [service load balancer](https://github.com/k3s-io/k3s/blob/master/pkg/servicelb/controller.go#L423) uses only the port number when creating a load balancer service container. If the MixedProtocolLBService feature-gate is enabled and an ingress is created with the same port in multiple protocols (e.g.: enabling access to port 53 on both UDP & TCP for DNS) this causes a name clash and the deployment fails.

A solution might be to include the protocol in the name (`fmt.Sprintf("lb-port-%d-%s", port.Port, port.Protocol)`) to avoid this clash as it is available to the function.

**Steps To Reproduce:**
- Installed K3s with arguments  `--disable traefik --kube-apiserver-arg feature-gates=MixedProtocolLBService=true`
- Install ingress-nginx and set TCP values `53: "${namespace}/${tcp-service}:53"` and UDP values `53: "${namespace}/${udp-service}:53"`

**Expected behavior:**
Load balancers should be created for each ingress route

**Actual behavior:**
The deployment fails due to a container name clash

**Additional context / logs:**
```
E1207 20:17:46.712564    3195 fieldmanager.go:191] "[SHOULD NOT HAPPEN] failed to update managedFields" err="failed to convert new object (apps/v1, Kind=DaemonSet) to smd typed: .spec.template.spec.containers: duplicate entries for key [name=\"lb-port-53\"]" VersionKind="/, Kind="
E1207 20:17:46.713208    3195 controller.go:135] error syncing 'ingress-nginx/ingress-controller-ingress-nginx-controller': handler svccontroller: failed to create ingress-nginx/svclb-ingress-controller-ingress-nginx-controller apps/v1, Kind=DaemonSet for svccontroller ingress-nginx/ingress-controller-ingress-nginx-controller: DaemonSet.apps "svclb-ingress-controller-ingress-nginx-controller" is invalid: spec.template.spec.containers[3].name: Duplicate value: "lb-port-53", requeuing
```

**Comment by brandond** [comment]
yeah, since `MixedProtocolLBService` is still Alpha we haven't updated the servicelb controller to support it yet. We will definitely need to do so before it goes to Beta.

**Comment by brandond** [comment]
This FeatureGate will go to Beta in 1.24, so we'll need to fix it by then.
https://github.com/kubernetes/enhancements/issues/1435

**Comment by katran001** [comment]
The next milestone is going to be v1.23.4 so as we approach v1.24.0, be on the lookout to get item closeout asap.

**Comment by rancher-max** [comment]
#### Validated using v1.23.5-rc1+k3s1

- I recreated this issue using v1.21.5+k3s2 but with simpler steps:
```
# Install k3s with the feature gate enabled
$ curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.21.5+k3s2 INSTALL_K3S_EXEC="server --cluster-init --write-kubeconfig-mode 644 --kube-apiserver-arg=feature-gates=MixedProtocolLBService=true" sh -

# Apply mixed-service manifest
$ k apply -f mixed.yaml

$ cat mixed.yaml 
apiVersion: v1
kind: Service
metadata:
  name: mixed-protocol
spec:
  type: LoadBalancer
  ports:
    - name: dns-udp
      port: 53
      protocol: UDP
    - name: dns-tcp
      port: 53
      protocol: TCP
  selector:
    app: my-dns-server
```
- Doing this on v1.21.5+k3s2 failed -- the service was stuck with ExternalIPs <pending> and there were no pods created by servicelb. Additionally, the same failure logs were seen as in this issue.
- **Doing these same steps but using v1.23.5-rc1+k3s1 succeeded: ExternalIPs are set and the servicelb pods are created. The logs do not have the same errors**

https://github.com/k3s-io/k3s/issues/5327 is seen, but that is not a blocking issue. We may fix that as part of this release, otherwise it will likely be fixed as part of next month's patch release.
